### PR TITLE
fix(ce-work): codify worktree isolation for parallel subagent dispatch

### LINKED ISSUE #682: feat(ce-work): codify worktree isolation for parallel subagent dispatch
## Problem

When `/ce:work` selects the \"parallel subagents\" execution strategy (Phase 1 Step 4), it dispatches multiple Agent tool calls for independent implementation units. Currently, the skill does not instruct the use of `isolation: \"worktree\"` on these Agent calls — subagents all operate on the same working directory, relying on a \"non-overlapping files\" heuristic.

This creates risks:
- Git operations can conflict between concurrent subagents
- Test runs can interfere with each other
- If the overlap assessment is wrong, file writes race

## Evidence: it already works when the model chooses it

In a real session (Opus, 3 plans executed in parallel), the orchestrating agent independently decided to use `isolation: \"worktree\"` on each Agent call. The result:
- 3 worktrees created at `.claude/worktrees/agent-{id}`, each on its own branch
- Subagents ran fully isolated with `run_in_background: true`
- Orchestrator reviewed diffs from all 3 worktrees, then merged sequentially
- 3 merge conflicts arose on overlapping files — resolved cleanly by the orchestrator

The flow worked well, but it was emergent behavior — the model happened to use the parameter. A different model or context could skip it entirely.

## Proposal

When the parallel subagents strategy is selected in Phase 1 Step 4, the skill should instruct:
1. Use `isolation: \"worktree\"` on each Agent dispatch
2. Use `run_in_background: true` for concurrent execution
3. After all subagents complete, review diffs from each worktree
4. Merge branches sequentially in dependency order, resolving conflicts
5. Clean up worktrees (note: `.claude/worktrees/` should be in `.gitignore` to avoid accidental staging)

This makes the parallel strategy reliable rather than model-dependent.

## Notes
- The serial subagents and inline strategies don't need this — they're sequential by definition
- PR #223 (`/lfg-ext`) explored a similar pattern for external delegates but hasn't been merged
- The git-worktree skill's `worktree-manager.sh` is an alternative to the Agent tool's built-in worktree support, but `isolation: \"worktree\"` is simpler since the harness manages the lifecycle automatically

### PR DESCRIPTION
**Pull request by tmchow**
## Summary

- Phase 1 Step 4's parallel-subagent flow described shared-directory dispatch with reactive collision recovery; worktree isolation only happened when the model spontaneously chose `isolation: "worktree"` (an emergent Opus behavior, not a skill instruction).
- Direct subagent dispatch to use `isolation: "worktree"` and `run_in_background: true` on Claude Code's `Agent` tool, with a documented fallback for platforms without built-in worktree primitives (Codex `spawn_agent`, Pi `subagent`).
- Reframe the "Parallel subagent constraints" block as the shared-directory fallback (no commits, no test runs by subagents) and add a new worktree-aware post-batch flow that resolves overlap at merge time instead of via the discovered-collision cross-check.

Stable/beta sync: applied identically to `ce-work` and `ce-work-beta` — the parallel-subagent body is shared verbatim between them.

## Test plan

- [x] `bun run release:validate` — passes (no manifest drift; doc-only edits)
- [x] `bun test` — 939 pass, 0 fail
- [x] Diff is two-file, +36/-8, scoped to Phase 1 Step 4 of both `SKILL.md` files

There is no code-level regression test for skill behavioral instructions; the verification is editorial review of the new isolation directive and the worktree-mode post-batch flow.

Fixes #682

[CE](https://github.com/EveryInc/compound-engineering-plugin)

### CODE REVIEW

File: plugins/compound-engineering/skills/ce-work/SKILL.md [hunk 1]
```diff
@@ -141,17 +141,22 @@ Determine how to proceed based on what was provided in `<input_document>`.
...
    - The specific unit's Goal, Files, Approach, Execution note, Patterns, Test scenarios, and Verification
    - Any resolved deferred questions relevant to that unit
    - Instruction to check whether the unit's test scenarios cover all applicable categories (happy paths, edge cases, error paths, integration) and supplement gaps before writing tests
 
-   **Parallel subagent constraints** — when dispatching units in parallel (not serial or inline):
+   **Shared-directory fallback constraints** — apply only when worktree isolation is unavailable:
    - Instruct each subagent: "Do not stage files (`git add`), create commits, or run the project test suite. The orchestrator handles testing, staging, and committing after all parallel units complete."
-   - These constraints prevent git index contention and test interference between concurrent subagents
+   - These constraints prevent git index contention and test interference between concurrent subagents.
+   - With worktree isolation active, omit these constraints — subagents may stage, commit, and run their unit's tests within their own worktree branch.
```
[the remark this answers is not in the document]
Comment (tmchow) [reply plugins/compound-engineering/skills/ce-work/SKILL.md:160]: > This new rule allows parallel subagents in worktree mode to `stage, commit, and run` tests, but Phase 2 still says parallel subagents do not commit and the orchestrator commits after the batch. [...] Please align both sections so commit ownership is explicitly split by mode (worktree-isolated vs shared-directory fallback).

Addressed: rewrote the Phase 2 Step 2 "Parallel subagent mode" note to split commit ownership by isolation mode — worktree-isolated subagents stage and commit inside their own worktree branch (orchestrator merges branches in dependency order after the batch); shared-directory fallback subagents do not commit (orchestrator stages and commits after the batch). The trailing applicability sentence now covers inline/serial execution, worktree-isolated subagents, and the orchestrator's post-batch commits in shared-directory mode. Both branches are now consistent with Phase 1 Step 4. Propagated identically to ce-work-beta. (commit 6cb4edba)

File: plugins/compound-engineering/skills/ce-work/SKILL.md [hunk 2]
```diff
@@ -162,7 +167,16 @@ Determine how to proceed based on what was provided in `<input_document>`.
...
    5. Dispatch the next unit
 
-   **After all parallel subagents in a batch complete:**
+   **After all parallel subagents in a batch complete (worktree-isolated mode):**
+   1. Wait for every subagent in the current parallel batch to finish.
+   2. For each completed subagent, in dependency order: review the worktree's diff against the orchestrator's branch. If the subagent did not commit its own work, stage and commit it inside that worktree.
+   3. Merge each subagent's branch into the orchestrator's branch sequentially in dependency order. Resolve conflicts at merge time — file overlap the Parallel Safety Check did not predict surfaces here, not as silent data loss.
+   4. After each merge, run the relevant test suite. If tests fail, diagnose and fix before merging the next branch.
+   5. Update the task list (progress is carried by the merge commits).
+   6. Remove each merged worktree (`git worktree remove .claude/worktrees/agent-<id>`).
```
[the remark this answers is not in the document]
Comment (tmchow) [reply plugins/compound-engineering/skills/ce-work/SKILL.md]: > Use a repo-root-resolved absolute path (for example via `git rev-parse --show-toplevel`) before calling `git worktree remove`.

Addressed: step 6 of the worktree-isolated post-batch flow now uses the absolute worktree path returned in the subagent's result as the primary form (the Claude Code Agent tool returns this when `isolation: "worktree"` is used and the subagent made changes), with `git rev-parse --show-toplevel`-based path construction as a fallback for platforms whose subagent primitive does not return a path. Cleanup now works regardless of the orchestrator's CWD. Propagated identically to ce-work-beta. (commit 6cb4edba)
